Changes to xSQLServerSetup (resolves issue #38)

johlju opened the pull request:
This PR
- Incorporates the changes from PR #46
- Also fixes issue #38

_This PR is based on the work by @jpmnteiro in the PR #46 which was abandon. Now @johlju is continuing the work to get this merged._

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xsqlserver/132)

<!-- Reviewable:end -->

johlju commented:
@mbreakey3 I guess you want a test for this one now, and code cleaned up according to style guide line? 😉

mbreakey3 commented:
Yes, that would be great 👍 :)

johlju commented:
I will fix that. Might take a few days because of Ignite. :)

mbreakey3 commented:
No problem!

kwirkykat commented:
@johlju @mbreakey3 What is happening here? Is this ready for review again?

mbreakey3 commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[README.md, line 351 at r1](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KUOTwA88anDlG28PSfo:-KUOTwA9-XXuY5R1hRHH:b-4haee3) ([raw file](https://github.com/powershell/xsqlserver/blob/6622426f8d7bcaff1faf49ed66b7ac609b760684/README.md#L351)):*
> ```Markdown
> ### Unreleased
> * Changes to xSQLServerSetup
>   - Now `Features` parameter is case-insensitive. 
> ```

Could you add a simple unit test for this change as well?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Just reviewed - label is correct now.

mbreakey3 commented:
@johlju - sorry, no rush on the unit test, just thought I'd put the comment in Reviewable.

johlju commented:
@mbreakey3 No worries. I feel a little bad that I haven't had time to finish the test yet, but I will. 😄  I have started, and my guess is that it is a day left of coding. I aiming at getting further with the test this weekend.
The weeks after MSIgnite has been busy. The little time I have been able to scrape together has been focused on helping others here with their reviews, to get things moving a bit more. 

Wish I could do this (code and review) full time instead of just in my spare time 😉  This is fun! 😄

mbreakey3 commented:
Glad you're enjoying it! Your contributions have been immensely helpful!

johlju commented:
@mbreakey3 I finally added a test. Yay! 😄  It do covers the changes in this PR (and a bit more).

But unfortunately there is more to be done to make this test cover all of the resource.
My thought is to submit an issue for the below remaining items. Reason for that is it would be good to get this test reviewed and merged so other PR's can add to this test an be merged as well.

* Need to test when SourceCredential is set. Must mock NetUse (Get-/Set-method) and Robocopy. _I submitted an issue to discuss if this functionality is necessary at all._
* Need to test when replication state is installed ($mockGetItemProperty_ConfigurationState is always set to `$null` now)
* Need to test other products than just DBENGINE.
* Need to test all parameters in Set-/Test-method
* Can't test Set-method when state is present because the Set-method is not currently built that way.

johlju commented:
@mbreakey3 the above comment isn't true anymore "It do covers the changes in this PR (and a bit more)".
After I did the style changes just now I did poke on lines that is not tested. If you want I will fix the test for that as well. I need to do test for basically the bullet points in the above comment 😉

mbreakey3 commented:
Reviewed 1 of 2 files at r4, 1 of 2 files at r5.
Review status: 2 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
I would say, open a new PR for adding the rest of the unit tests for this resource, as your tests cover the main changes that this PR is fixing.

johlju commented:
@mbreakey3 I raised the code coverage of the test from 60% to 88%. I submitted a issue for the rest. So I can get back to writing the last tests when this PR has merged. Or, if the review takes a while. I might have time to get the last tests in as well. 😄

mgreenegit commented:
Reviewed 2 of 3 files at r7, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[xSQLServerHelper.psm1, line 61 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90SnI54EWl8v3RIhw:-KW90SnI54EWl8v3RIhx:bjnupb1) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/xSQLServerHelper.psm1#L61)):*
> ```PowerShell
> }
> 
> function Connect-SQLAnalysis
> ```

Please add help information

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 2 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90YPO9Y8lcGLN_VE-:-KW90YPO9Y8lcGLN_VE0:b7542si) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L2)):*
> ```PowerShell
> $script:currentPath = Split-Path -Parent $MyInvocation.MyCommand.Path
> Import-Module $script:currentPath\..\..\xSQLServerHelper.psm1 -ErrorAction Stop
> ```

Add named param

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 3 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90_9MkR3N118bHrj0:-KW90_9MkR3N118bHrj1:b7542si) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L3)):*
> ```PowerShell
> $script:currentPath = Split-Path -Parent $MyInvocation.MyCommand.Path
> Import-Module $script:currentPath\..\..\xSQLServerHelper.psm1 -ErrorAction Stop
> Import-Module $script:currentPath\..\..\xPDT.psm1
> ```

Add named param

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 5 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90aVD_ZVdA6NZ0Fw_:-KW90aVD_ZVdA6NZ0Fwa:b-2zder3) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L5)):*
> ```PowerShell
> Import-Module $script:currentPath\..\..\xPDT.psm1
> 
> function Get-TargetResource
> ```

Add help information

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 48 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90iIMT-EMdIK1oZ13:-KW90iIMT-EMdIK1oZ14:bnn8cc1) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L48)):*
> ```PowerShell
>     }
>     
>     if ($InstanceName -eq 'MSSQLSERVER')
> ```

These could be aligned (style recommendation)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 84 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90sPyzafGxhkLIaDV:-KW90sPyzafGxhkLIaDW:b7542si) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L84)):*
> ```PowerShell
>         if ($isReplicationInstalled -eq 1)
>         {
>             Write-Verbose 'Replication feature detected'
> ```

Add named param

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 89 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90tt3uIRBSkOHKd9n:-KW90tt3uIRBSkOHKd9o:b7542si) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L89)):*
> ```PowerShell
>         else
>         {
>             Write-Verbose 'Replication feature not detected'
> ```

Add named param

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 253 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW914URz4GSy6ep51sD:-KW914URz4GSy6ep51sE:b-lvehya) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L253)):*
> ```PowerShell
>     }
> 
>     return @{
> ```

This would be a great opportunity to align the items in this hashtable

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 287 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW919arbi714JgfeOM7:-KW919arbi714JgfeOM8:b-2zder3) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L287)):*
> ```PowerShell
> }
> 
> function Set-TargetResource
> ```

Add help information

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 419 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91C9F5QASMw8ocsxu:-KW91C9F5QASMw8ocsxv:beovvro) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L419)):*
> ```PowerShell
>     )
> 
>     $parameters = @{
> ```

Align these on the "="

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 766 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91S0n2zrh2OZ3n2M-:-KW91S0oT8LfU4DHIfeP:bwi5vio) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L766)):*
> ```PowerShell
> }
> 
> function Test-TargetResource
> ```

Would you mind adding help information?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 899 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91Y1UeJ7b1A7Lp6oR:-KW91Y1UeJ7b1A7Lp6oS:b2lvjpy) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L899)):*
> ```PowerShell
>     )
> 
>     $parameters = @{
> ```

Align on "-"

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 908 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91_I811voXVl1qmCU:-KW91_I811voXVl1qmCV:b7542si) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L908)):*
> ```PowerShell
> 
>     $sqlData = Get-TargetResource @parameters
>     Write-Verbose "Features found: '$($SQLData.Features)'"
> ```

Add named param

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 922 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91bu0mREr6sdif3Ru:-KW91bu0mREr6sdif3Rv:b7542si) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L922)):*
> ```PowerShell
>             if(!($sqlData.Features.Contains($feature)))
>             {
>                 Write-Verbose "Unable to find feature '$feature' among the installed features: '$($sqlData.Features)'"
> ```

Add named param

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 931 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91emQemVIUY69hpzS:-KW91emQemVIUY69hpzT:b-ce9nf1) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L931)):*
> ```PowerShell
> }
> 
> function GetSQLVersion
> ```

Would you mind adding help information for this one?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 944 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91hO11qJepTJ3LlVL:-KW91hO11qJepTJ3LlVM:bjnupb1) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L944)):*
> ```PowerShell
> }
> 
> function Get-FirstItemPropertyValue
> ```

Please add help information

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 967 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91kbocd9n_ZtvE1_R:-KW91kbocd9n_ZtvE1_S:bjnupb1) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L967)):*
> ```PowerShell
> }
> 
> function Copy-ItemWithRoboCopy
> ```

Please add help information

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 984 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91metOVrJo6oeYfpJ:-KW91metOVrJo6oeYfpK:bjnupb1) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L984)):*
> ```PowerShell
> }
> 
> function Get-TemporaryFolder
> ```

Please add help information

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[README.md, line 351 at r1](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KUOTwA88anDlG28PSfo:-KWcke_4ggcnJoubdOV6:b-6ukefh) ([raw file](https://github.com/powershell/xsqlserver/blob/6622426f8d7bcaff1faf49ed66b7ac609b760684/README.md#L351)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Could you add a simple unit test for this change as well?
</details>
@mbreakey3 I'm sorry, I don't follow you. Happy to do unit tests, but for what do you want me to do a unit test? The line you comment on is an already present test. :)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 48 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90iIMT-EMdIK1oZ13:-KWd6WYkgTZqgvpI2Y24:boxjhwh) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L48)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> These could be aligned (style recommendation)
</details>
@mgreenegit I rather not do this until what you suggest is in the style guidelines. I think we can only enforce style changes that can be backed up by the style guideline. I we change this in the style guideline, then I'm happy to enforce this also. Until then I'm humbly decline changing this. :smile: 

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 253 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW914URz4GSy6ep51sD:-KWd7La3VTiEQXZ1WsXj:b-8ewxk5) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L253)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> This would be a great opportunity to align the items in this hashtable
</details>
See previous comment regarding this.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 419 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91C9F5QASMw8ocsxu:-KWd7OcU4zajdLWX3adk:b-8ewxk5) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L419)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Align these on the "="
</details>
See previous comment regarding this.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 899 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91Y1UeJ7b1A7Lp6oR:-KWd7PXUuah_ySoJrYB9:b-8ewxk5) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L899)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Align on "-"
</details>
See previous comment regarding this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 1 of 2 files at r8.
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[README.md, line 351 at r1](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KUOTwA88anDlG28PSfo:-KWdL7jDURBl0uO-FNqI:b67o9mu) ([raw file](https://github.com/powershell/xsqlserver/blob/6622426f8d7bcaff1faf49ed66b7ac609b760684/README.md#L351)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> @mbreakey3 I'm sorry, I don't follow you. Happy to do unit tests, but for what do you want me to do a unit test? The line you comment on is an already present test. :)
</details>
Yeah, this looks fine.... not sure what I was referring to here. Must have missed it.

---

*[README.md, line 371 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KWdKhvEeDGPFnLKQ5TC:-KWdKhvEeDGPFnLKQ5TD:b-bjwlho) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/README.md#L371)):*
> ```Markdown
>    - 1-SetDatabaseOwner.ps1
>  * Added tests for resources
>    - MSFT_xSQLServerDatabaseOwner.Tests.Tests.ps1
> ```

Just one '.Tests' here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

johlju commented:
@mgreenegit appreciate the review! Please look this over again if you have a chance.

---

Reviewed 1 of 3 files at r7, 4 of 4 files at r9.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[README.md, line 371 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KWdKhvEeDGPFnLKQ5TC:-KWdZueCrphMl-zMSOnw:bvmf93n) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/README.md#L371)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Just one '.Tests' here.
</details>
Removed 'Tests' altogether since we list the tests (and other tests) under the list item 'Added test for resource'.

---

*[xSQLServerHelper.psm1, line 61 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90SnI54EWl8v3RIhw:-KWd_B7dqwYwkQ9CwFa1:b-c3evwk) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/xSQLServerHelper.psm1#L61)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Please add help information
</details>
I added comment-based help for all of the functions in this module.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 2 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90YPO9Y8lcGLN_VE-:-KWd_IvNfAm6YhtFJfa2:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L2)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 3 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90_9MkR3N118bHrj0:-KWd_KPfWpIu7Hln0KMp:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L3)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 5 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90aVD_ZVdA6NZ0Fw_:-KWd_LwdPfAcS8lRnesO:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L5)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add help information
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 84 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90sPyzafGxhkLIaDV:-KWd_N6wV5K93dljAnpt:bc74bh7) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L84)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done. And also fixed all `Write-Verbose` to use the helper function `New-VerboseMessage` and made sure they does not use positional parameters.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 89 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90tt3uIRBSkOHKd9n:-KWd_dZgDmCbTgzD9rr0:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L89)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 287 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW919arbi714JgfeOM7:-KWd_fC42mVJlWXWFKIM:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L287)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add help information
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 766 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91S0n2zrh2OZ3n2M-:-KWd_gKtgH5WixE-WsRE:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L766)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Would you mind adding help information?
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 908 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91_I811voXVl1qmCU:-KWd_hZvfnW-bKBrLTtF:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L908)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 922 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91bu0mREr6sdif3Ru:-KWd_icwrDR9D0tKPKSc:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L922)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 931 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91emQemVIUY69hpzS:-KWd_jo3yF9QSVpqmkdC:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L931)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Would you mind adding help information for this one?
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 944 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91hO11qJepTJ3LlVL:-KWd_kqqrLrQr-B8u7w2:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L944)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Please add help information
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 967 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91kbocd9n_ZtvE1_R:-KWd_lynvp0021KNqiKS:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L967)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Please add help information
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 984 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91metOVrJo6oeYfpJ:-KWd_n7k4Oj_ZwHdipVa:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L984)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Please add help information
</details>
Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-:-KWd_yIPed_ek86_TZ0z:b4to66e)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 1 of 4 files at r9.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

luigilink commented:
@johlju You have the same issues than me.
AppVeyor or PSA will updated, maybe ?

johlju commented:
Yes, this went thru AppVeyor yesterday. must be new test code. Asked @kwirkykat to look into it. Will file issue in repo also.

johlju commented:
Unless there are no more review comments on this by Monday evening, I will merge this PR. We have other PR's waiting for the tests in this PR.

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 1 files at r10, 2 of 2 files at r11.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[README.md, line 371 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KWdKhvEeDGPFnLKQ5TC:-KX6eH_f2i71LjTMY035:bb6ywmx) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/README.md#L371)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Removed 'Tests' altogether since we list the tests (and other tests) under the list item 'Added test for resource'.
</details>
LGTM

---

*[xSQLServerHelper.psm1, line 61 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90SnI54EWl8v3RIhw:-KX6eX1D9-tPlBgYsyRq:bb6ywmx) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/xSQLServerHelper.psm1#L61)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> I added comment-based help for all of the functions in this module.
</details>
LGTM

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 2 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90YPO9Y8lcGLN_VE-:-KX6eflW08e5M00a1FmU:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L2)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 3 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90_9MkR3N118bHrj0:-KX6emsnOWxAJa73Zafp:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L3)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 5 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90aVD_ZVdA6NZ0Fw_:-KX6eoi98yxdO7Kqh9zz:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L5)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 48 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90iIMT-EMdIK1oZ13:-KX6eptoSJHIg6IjNFzH:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L48)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> @mgreenegit I rather not do this until what you suggest is in the style guidelines. I think we can only enforce style changes that can be backed up by the style guideline. I we change this in the style guideline, then I'm happy to enforce this also. Until then I'm humbly decline changing this. :smile: 
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 84 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90sPyzafGxhkLIaDV:-KX6erChYri2SJUYQmeB:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L84)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done. And also fixed all `Write-Verbose` to use the helper function `New-VerboseMessage` and made sure they does not use positional parameters.
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 89 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90tt3uIRBSkOHKd9n:-KX6esCA8VaHtDcvOFIB:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L89)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 253 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW914URz4GSy6ep51sD:-KX6etKYRqjUR-JTbMt_:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L253)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> See previous comment regarding this.
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 287 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW919arbi714JgfeOM7:-KX6euK8BxOpjgSijX4e:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L287)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 419 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91C9F5QASMw8ocsxu:-KX6evBsFlRKcXYNWI-H:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L419)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> See previous comment regarding this.
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 766 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91S0n2zrh2OZ3n2M-:-KX6ewTZpWNtPvuskTNP:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L766)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 899 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91Y1UeJ7b1A7Lp6oR:-KX6exJHLDY9iTNhupZe:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L899)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> See previous comment regarding this.
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 908 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91_I811voXVl1qmCU:-KX6eyFg4gnkYEHmMuUq:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L908)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 922 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91bu0mREr6sdif3Ru:-KX6ez2ccINEa_bZDmur:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L922)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 931 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91emQemVIUY69hpzS:-KX6f-2e5UT3OEhx3qq7:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L931)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 944 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91hO11qJepTJ3LlVL:-KX6f-sZC9qOVS1SN6TS:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L944)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 967 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91kbocd9n_ZtvE1_R:-KX6f1-nsFDnMT9MtBev:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L967)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 984 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW91metOVrJo6oeYfpJ:-KX6f1ngI5bYha7Fqi_g:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L984)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-:-KX6f42D7gLrG2yQ9v79:bg7igkv)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[xSQLServerHelper.psm1, line 61 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90SnI54EWl8v3RIhw:-KX6fzuEaud-VroZa7bu:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/xSQLServerHelper.psm1#L61)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> LGTM
</details>
Done

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 2 at r8](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KW90YPO9Y8lcGLN_VE-:-KX6gsibQ_pOL2et5VGo:bb6ywmx) ([raw file](https://github.com/powershell/xsqlserver/blob/cbbd5e6a8be21f49919f8ce0cc526bf38c54ed6f/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L2)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done.
</details>
LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

johlju commented:
@mgreenegit Could you please go an Acknowledge your review comments?

johlju commented:
Review status: all files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[README.md, line 359 at r11](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KX9xzV8jAfHq5qMPqsW:-KX9xzV8jAfHq5qMPqsX:b3i54kl) ([raw file](https://github.com/powershell/xsqlserver/blob/085dee5172bdacbfdb0885ccbca5475e59889171/README.md#L359)):*
> ```Markdown
> * Fixes in xSQLAOGroupJoin
>   - Availability Group name now appears in the error message for a failed Availability Group join attempt.
> * Changes to xSQLServerSetup
> ```

The Unreleased section has been moved to the CHANGELOG.md. Please move your entries to the Unreleased section in the new CHANGELOG.md when you do `git rebase`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 1 of 2 files at r12, 2 of 2 files at r13.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[README.md, line 359 at r11](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-KX9xzV8jAfHq5qMPqsW:-KXCKnq_0ARafzzdts6N:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/085dee5172bdacbfdb0885ccbca5475e59889171/README.md#L359)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> The Unreleased section has been moved to the CHANGELOG.md. Please move your entries to the Unreleased section in the new CHANGELOG.md when you do `git rebase`.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Review status: all files reviewed at latest revision, 19 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-:-KXCL1423pPTzsg6sq_n:bg7igkv)*
<!-- Sent from Reviewable.io -->

johlju commented:
There are conflicts that needs to be resolved on this one.

johlju commented:
<img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r14.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/132#-:-KXFXmdPcfg1aWVUHfCt:bmxilld)*
<!-- Sent from Reviewable.io -->
